### ISSUE TITLE
Serialization issue with LinearRing

### ISSUE THREAD

**Issue by nicoulaj** [Toblerity/Shapely#547]
## Expected behavior and actual behavior.

When pickling a LinearRing and trying to build a Polygon from it, the Polygon is not initialized, with a trace from geos: `Shell is not a LinearRing`. Attempting to run boolean operations results in `ValueError: Null geometry supports no operations`.

## Steps to reproduce the problem.

```python
def test_pickle_linear_ring():
    import pickle
    from shapely.geometry import Polygon, LinearRing

    ring = LinearRing([(1, 1), (2, 1), (2, 2), (1, 2), (1, 1)])
    assert Polygon(ring).difference(Polygon(ring)).is_empty

    pickled = pickle.loads(pickle.dumps(ring))
    assert Polygon(pickled).difference(Polygon(pickled)).is_empty
```

## Operating system
Fedora 27, x86_64

## Shapely version and provenance

- `Shapely[vectorized]==1.6.2.post1` installed from PyPI using pip
- geos 3.6.1
- Python 3.6.3

**Comment by snorfalorpagus** [comment]
I can reproduce this. It looks like the issue is caused by the lack of a distinction between LinearRing/LinearString in the WKB format that is used to pickle the geometry.

https://en.wikipedia.org/wiki/Well-known_text#Well-known_binary

```
>>> ring.wkt
'LINEARRING (1 1, 2 1, 2 2, 1 2, 1 1)'
>>> pickled.wkt
'LINESTRING (1 1, 2 1, 2 2, 1 2, 1 1)'
>>> ring.wkb == pickled.wkb
True
>>> lgeos.GEOSGeomType(ring._geom)
b'LinearRing'
>>> lgeos.GEOSGeomType(pickled._geom)
b'LineString'
```

We could probably work around this by modifying the unpickle method to check the class type. If the class is a `LinearRing` we can create a new geometry around the existing coordinate sequence.

**Comment by snorfalorpagus** [comment]
Closed by #548.

**Comment by nicoulaj** [comment]
Thanks for the fast fix :)
